eclair-core/src/main/scala/fr/acinq/eclair/json/JsonSerializers.scala

```diff
@@ -203,9 +230,12 @@ object OutPointKeySerializer extends MinimalKeySerializer({
   case x: OutPoint => s"${x.txid}:${x.index}"
 })
 
-object InputInfoSerializer extends MinimalSerializer({
-  case x: InputInfo => JObject(("outPoint", JString(s"${x.outPoint.txid}:${x.outPoint.index}")), ("amountSatoshis", JInt(x.txOut.amount.toLong)))
-})
+// @formatter:off
+private case class InputInfoJson(outPoint: String, amountSatoshis: Long)
+object InputInfoSerializer extends ConvertClassSerializer[InputInfo](i =>
+  InputInfoJson(outPoint = s"${i.outPoint.txid}:${i.outPoint.index}", amountSatoshis = i.txOut.amount.toLong)
+)
+// @formatter:on
```

Comment:
Since we already have a serializer for `OutPoint` and one for `MilliSatoshi`, it feels like we should be able to do:

```scala
private case class InputInfoJson(outPoint: OutPoint, amountSatoshis: Satoshi)
object InputInfoSerializer extends ConvertClassSerializer[InputInfo](i => InputInfoJson(i.outPoint, i.txOut.amount))
```

Can't we?

Reply:
Absolutely! And this is indeed a perfect show case of this new technique. Done in 62c9e4151.